File: flutter_app/android/app/src/main/java/com/flutter_app/FlutterChannelPlugin.java
package com.flutter_app;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.flutter.Log;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.BasicMessageChannel;
import io.flutter.plugin.common.EventChannel;
import io.flutter.plugin.common.MessageCodec;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.StandardMessageCodec;

public class FlutterChannelPlugin implements FlutterPlugin {

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
        //1
        MethodChannel methodChannel = new MethodChannel(
                binding.getBinaryMessenger(),
                "flutter/live/methodChannel");
        methodChannel.setMethodCallHandler(new FlutterMethodChannel());

        //2
        EventChannel eventChannel = new EventChannel(
                binding.getBinaryMessenger(),
                "flutter/live/eventChannel");
        eventChannel.setStreamHandler(new FlutterStreamChannel());

        //3
        BasicMessageChannel messageChannel = new BasicMessageChannel(
                binding.getBinaryMessenger(),
                "flutter/live/messageChannel",
                new StandardMessageCodec()
        );
        messageChannel.setMessageHandler(new FlutterMessageChannel());
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {

    }

    //MethodChannel 协议实现部分
    public class FlutterMethodChannel implements MethodChannel.MethodCallHandler {
        @Override
        public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
            Log.e("FlutterMethodChannel",call.toString());
        }
    }

    //EventChannel 协议实现部分
    public class FlutterStreamChannel implements EventChannel.StreamHandler {
        @Override
        public void onListen(Object arguments, EventChannel.EventSink events) {

        }

        @Override
        public void onCancel(Object arguments) {

        }
    }

    //MessageChannel 协议实现部分
    public  class FlutterMessageChannel implements BasicMessageChannel.MessageHandler {
        @Override
        public void onMessage(@Nullable Object message, @NonNull BasicMessageChannel.Reply reply) {
            Log.e("FlutterMessageChannel",message.toString());
        }
    }
}

File: flutter_app/ios/Runner/FlutterChannelPlugin.m
//
//  FlutterChannelPlugin.m
//  Runner
//
//  Created by W E on 2020/8/10.
//

/*
不能直接这样在ios项目中写插件的回调，
因为每次运行都会更新GeneratedPluginRegistrant 这个类，
会按podfile的描述重新生成，
1、要不就统一像这样直接写
2、要不就统一按插件规范写
3、要不就在AppDelegate里边写，（这里不会重新生成）
 
1、MethodChannel 
2、EventChannel //Stream数据
3、MessageChannel
*/

#import "FlutterChannelPlugin.h"

@interface FlutterChannelPlugin()
@property (nonatomic) FlutterEventSink eventSink;
@end


@implementation FlutterChannelPlugin

//------------------------------FlutterPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterChannelPlugin *plugin = [[FlutterChannelPlugin alloc] init];
    [plugin createMethodChannel:registrar];
    [plugin createEventChannel:registrar];
    [plugin createMessageChannel:registrar];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {

  if ([call.method isEqualToString:@"startLive"]) {
      NSLog(@"Flutter MethodChannel 收到:startLive");
  }
  else {
    result(FlutterMethodNotImplemented);
  }
}

//------------------------------MethodChannel

- (void)createMethodChannel:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* methodChannel = [FlutterMethodChannel
    methodChannelWithName:@"flutter/live/methodChannel"
          binaryMessenger:[registrar messenger]];
    [registrar addMethodCallDelegate:self channel:methodChannel];
}


//------------------------------EventChannel && FlutterStreamHandler

- (void)createEventChannel:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterEventChannel *eventChannel = [FlutterEventChannel eventChannelWithName:@"flutter/live/eventChannel" binaryMessenger:[registrar messenger]];
    [eventChannel setStreamHandler:self];
}

- (FlutterError * _Nullable)onCancelWithArguments:(id _Nullable)arguments {
    _eventSink = nil;
    return nil;
}

- (FlutterError * _Nullable)onListenWithArguments:(id _Nullable)arguments eventSink:(nonnull FlutterEventSink)events {
    _eventSink = events;
    return nil;
}


//------------------------------MessageChannel

- (void)createMessageChannel:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterBasicMessageChannel *messageChannel = [FlutterBasicMessageChannel messageChannelWithName:@"flutter/live/messageChannel" binaryMessenger:[registrar messenger]];
    [messageChannel setMessageHandler:^(id  _Nullable message, FlutterReply  _Nonnull callback) {
        NSLog(@"MessageChannel 收到：%@",message);
//        NSString *method=message[@"method"];
//        if ([method isEqualToString:@"startLive"]) {
//            NSLog(@"Flutter MessageChannel 收到:startLive");
//        }
    }];
}

@end
